### PR TITLE
Fix search query splitting

### PR DESCRIPTION
**Pull request by cryogenian**

### CODE REVIEW

File: test/Test/S.purs [hunk 1]
```diff
@@ -63,7 +64,9 @@ searchTest = do
             {include: true, labels: [Common "foo"],
              predicate: Contains (Text "uni*")},
             {include: true, labels: [Meta "path"],
-             predicate: Contains (Text "/foo/bar")}
+             predicate: Contains (Text "/foo/bar")},
+            {include: true, labels: [Common "path"],
+             predicate: Contains (Text "\"foo bar\"")}
```
Comment (puffnfresh) [inline test/Test/S.purs:69]: Should the quotes be ignored/skipped?

Comment (jdegoes) [inline test/Test/S.purs:69]: Good point. These quotes are purely for the parser to allow special characters, so they shouldn't appear in the ADT. @cryogenian Can you do a separate PR to fix this?
